src/Writer.php

```diff
@@ -226,6 +226,9 @@ public function addValidator(callable $validator, string $validator_name): self
      */
     public function setNewline(string $newline): self
     {
+        if ("\n" !== $this->newline && $this->document instanceof Stream && !$this->document->isSeekable()) {
```

Comment:
I don't think you need to add this check here... not being able to seek should not prevent setting the newline. This is to be consistent with for instance providing a non writable CSV document to the Writer class. on its own it is not forbidden but you'll get an error on insertion. It should be the same with a non seekable document.